server/data-service/src/main/java/edu/illinois/ncsa/incore/service/data/utils/GeoserverRestApi.java

```diff
@@ -140,51 +140,78 @@ public boolean postFileToGeoserver(String apiUrl, File inData, String contType)
-            // to rename, the POST action should create datastore first with the geopackage data,
-            // then create layer with xml to control the name to be changed to dataset id.
-            // if you don't need to rename, set renameData boolean in following line to false
-            Boolean renameData = true;
-            if (renameData) {
-                // create datastore then create layer
-                try {
-                    String restUrl = this.geoserverUrl + "/rest";
-                    String fileNameNoExt = fileName.split("\\.")[0];
-                    int datastoreResponse = createDatastore(restUrl, GEOSERVER_WORKSPACE, store, inFile.getAbsolutePath(), "geopackage");
-                    int layerResponse = createLayer(restUrl, GEOSERVER_WORKSPACE, store, store, fileNameNoExt);
-                    if (datastoreResponse == 201 && layerResponse == 201) {
-                        published = true;
-                    }
-                } catch (IOException e) {
-                    throw new RuntimeException(e);
-                }
+            if (renameLayer) {
+                published = this.uploadToGeoserverWithRenaming(fileName, store, inFile, "geopackage");
             } else{
                 String restUrl = this.geoserverUrl + "/rest/workspaces/" + GEOSERVER_WORKSPACE + "/datastores/" + store +
                     "/file.gpkg?configure=all&name=" + store;
                 published = this.postFileToGeoserver(restUrl, inFile, "gpkg");
             }
         } else if (inExt.equalsIgnoreCase("tif")) {
+            // currently, renaming of the raster layer is not being supported
             String restUrl = this.geoserverUrl + "/rest/workspaces/" + GEOSERVER_WORKSPACE + "/coveragestores/" + store + "/file.geotiff";
             published = this.postFileToGeoserver(restUrl, inFile, "tif");
         }
 
         return published;
     }
 
+    /**
+     * upload file to geoserver with renaming
+     * this method is for the case when the file name is different from published layer name
+     *
+     * @param fileName
+     * @param store
+     * @param inFile
+     * @param fileForamt
+     * @return
+     */
+    public Boolean uploadToGeoserverWithRenaming(String fileName, String store, File inFile, String fileForamt) {
+        Boolean published = false;
+        try {
+            String restUrl = this.geoserverUrl + "/rest";
+            String fileNameNoExt = fileName.split("\\.")[0];
+            int datastoreResponse = createDatastore(restUrl, GEOSERVER_WORKSPACE, store, inFile.getAbsolutePath(), fileForamt);
+            int layerResponse = createLayer(restUrl, GEOSERVER_WORKSPACE, store, store, fileNameNoExt);
+            if (datastoreResponse == 201 && layerResponse == 201) {
+                published = true;
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
Maybe add a logger.error here so we capture the log?

Reply:
added logger.error